### PR TITLE
🔒️ do asset transfers before processDeposit

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

**Review by acollette — APPROVED** [review]

File: src/Registry.sol [hunk 1]
```diff
@@ -409,17 +421,37 @@ contract Registry is IRegistry, RegistryGuardian {
...
+     * 3 = ERC1155.
+     * ...
+     */
+    function batchGetAssetTypes(address[] calldata assetAddresses)
+        external
+        view
+        returns (uint256[] memory assetTypes)
+    {
+        uint256 addrLength = assetAddresses.length;
+        assetTypes = new uint256[](addrLength);
```
Comment (j-vp) [inline src/Registry.sol:440]: is this one needed, since it's already defined in the return?

File: src/Registry.sol [hunk 2]
```diff
@@ -487,17 +518,16 @@ contract Registry is IRegistry, RegistryGuardian {
         address assetAddress;
         if (creditor == address(0)) {
             for (uint256 i; i < addrLength; ++i) {
-                assetAddress = assetAddresses[i];
```
Comment (j-vp) [inline src/Registry.sol:490]: comment related to L513 (return): do we need to return it here? or should we do we do a similar thing as in deposits and first fetch the asset types?

File: src/Registry.sol (hunk 1 above)
Comment (Thomas-Smets) [reply src/Registry.sol:440]: Yes, the assetTypes defined in the return has length 0.

File: src/Registry.sol (hunk 2 above)
Comment (Thomas-Smets) [reply src/Registry.sol:490]: We get it for free here, why would we get via an additional loop over all asset modules?

**Review by j-vp — APPROVED** [review]
gtg

**Pull request merged**